### PR TITLE
Add nrFiltered attribute to namespaceSample when filtering by namespace

### PR DESCRIPTION
**Pull request by alvarocabanas**
Signed-off-by: Alvaro Cabanas <acabanas@newrelic.com>

### CODE REVIEW

File: src/definition/populate.go [hunk 1]
```diff
@@ -65,7 +67,10 @@ func IntegrationPopulator(config *IntegrationPopulateConfig) (bool, []error) {
 				if nsGetter := config.Specs[groupLabel].NamespaceGetter; nsGetter != nil {
 					ns := nsGetter(metrics)
 					if !config.Filterer.IsAllowed(ns) {
-						continue
+						if groupLabel != NamespaceGroup {
```
Comment (alvarocabanas) [inline src/definition/populate.go:70]: Here we are breaking the Genericness of the function, but it was the best way we found to add this label

**Review by paologallinaharbur — APPROVED** [review]

File: src/metric/definition.go [hunk 2]
```diff
@@ -727,9 +727,8 @@ var KSMSpecs = definition.SpecGroups{
 		},
 	},
 	"hpa": {
-		IDGenerator:     prometheus.FromLabelValueEntityIDGenerator("kube_hpa_labels", "hpa"),
-		TypeGenerator:   prometheus.FromLabelValueEntityTypeGenerator("kube_hpa_labels"),
-		NamespaceGetter: prometheus.FromLabelGetNamespace,
+		IDGenerator:   prometheus.FromLabelValueEntityIDGenerator("kube_hpa_labels", "hpa"),
+		TypeGenerator: prometheus.FromLabelValueEntityTypeGenerator("kube_hpa_labels"),
```
Comment (paologallinaharbur) [inline src/metric/definition.go:731]: horizontalpodautoscalers          hpa          autoscaling/v1                         true         HorizontalPodAutoscaler

Comment (alvarocabanas) [reply src/metric/definition.go:731]: I've added the filtering because HPA are indeed namespaced as Paolo pointed out from the kubectl api-resources call.

**Review by marcsanmi — APPROVED** [review]
LGTM 👍

**Pull request merged**